library/Opus/Model2/AbstractModel.php

```diff
@@ -117,6 +117,22 @@ protected static function getRepository()
     /** @return int|string */
     abstract public function getId();
 
+    // TODO Should we again allow preStore() to return a non-null value
+    //      (in which case the storage process would be canceled)?
+    /**
+     * Perform any actions needed to provide storing.
+     */
+    protected function preStore()
```

Comment:
Die Funktionen sollten weiterhin einen Return-Wert haben, auch wenn wir ihn im Augenblick nicht auswerten.

Reply:
Wie vorher sollte der "default" alles ist gut Wert **null** sein.

Reply:
Ok, und sollte die `store()` Methode (wie zuvor) wirklich auch diesen `non-null` Wert zurückgeben? Also so:
```
    /**
     * Persist all the models information to its database locations.
     *
     * @return int ID of stored object
     * @throws DbException
     */
    public function store()
    {
        $pre = $this->preStore();
        if ($pre !== null) {
            return $pre;
        }
```
Es ist ja nicht gesagt, dass der Rückgabewert von `preStore()` die ID eines in der Datenbank existierenden Objekts ist, oder?

Reply:
Nein, ich denke nicht, `preStore` und `postStore` sind Konzepte innerhalb des Persistence-Layers. Die Code der `store` aufruft muss/soll davon nichts wissen.

Reply:
Ok, was würde `store()` dann im Fall eines Abbruchs zurückgeben? `0`?
```
    /**
     * Persist all the models information to its database locations.
     *
     * @return int ID of stored object
     * @throws DbException
     */
    public function store()
    {
        $pre = $this->preStore();
        if ($pre !== null) {
            return 0;
        }
```
Oder bräuchte es dann wieder einen `mixed|null` return value für `store()`?

Reply:
Momentan gibt es keine Abbrüche. Wir wissen ja noch nicht mal, ob der Code so bleibt. Vielleicht werfen wir in Zukunft Exceptions. Im Zweifelsfall soll sich der Code einfach so verhalten, wie vorher, vielleicht mit einem Vermerk ("TODO DOCTRINE"), falls "vorher" Probleme hat. Danke!